packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -82,56 +42,141 @@ private float getTextXOffset(float markerWidth, Rect rect) {
         return (markerWidth / 2f) - (rect.width() / 2f) - rect.left;
     }
 
-    private Path addTailOnMarkerCenter(Bitmap marker, int tailSize, int shadowSize) {
+    private Path addPointerOnMarkerCenter(Bitmap marker, int pointerWidth, int pointerHeight, int shadowSize) {
         Path pointer = new Path();
         pointer.setFillType(Path.FillType.EVEN_ODD);
         float width = marker.getWidth();
-        float height = marker.getHeight() - tailSize - shadowSize;
-        pointer.moveTo(width / 2f - tailSize, height);
-        pointer.lineTo(width / 2f + tailSize, height);
-        pointer.lineTo(width / 2f, height + tailSize);
-        pointer.lineTo(width / 2f - tailSize, height);
+        float height = marker.getHeight() - pointerHeight - shadowSize;
+        pointer.moveTo(width / 2f - pointerWidth, height);
+        pointer.lineTo(width / 2f + pointerWidth, height);
+        pointer.lineTo(width / 2f, height + pointerHeight);
+        pointer.lineTo(width / 2f - pointerWidth, height);
         pointer.close();
         return pointer;
     }
 
-    private Bitmap getMarkerBitmap(String text, int markerColor, int textColor, boolean hasTail) {
+    private Bitmap getIconBitmap(String svgIcon, int width, int height) {
+        if(svgIcon == null)
+            return null;
+        
+        final Bitmap bitmap = markerCache.getBitmapFromMemCache(svgIcon);   
+        if (bitmap != null) {
+            return bitmap;
+        }
+
+        try{
+            //Needed this because svg doesn't scale if there is fixed width and height
+            String finalSvgIcon = svgIcon.replaceAll("width=\"24\" height=\"24\" ", "");
```

Comment:
Shouldn't we use a more generic regex here? Isn't it possible that the svg has different width/height?

Reply:
Thanks, I just implemented this in another branch because I needed to render the chevron icon haha. I will commit it here